projects/npm-tools/packages/npm-scripts/src/utils/getJestModuleNameMapper.js

```diff
@@ -80,20 +80,42 @@ function getJestModuleNameMapper() {
 					if (main) {
 						const entry = path.join(project, ...SRC_PATH, main);
 
-						if (fs.existsSync(entry)) {
-							const resources = path.relative(
-								cwd,
-								path.join(project, ...SRC_PATH)
-							);
-
-							mappings[`^${name}$`] = `<rootDir>${path.relative(
-								cwd,
-								entry
-							)}`;
-
-							mappings[
-								`^${name}/(.*)`
-							] = `<rootDir>${resources}/$1`;
+						// Handle typical formats for "main":
+						//
+						// - index        -> index.js
+						// - index.es     -> index.es.js
+						// - index.es.js  -> index.es.js
+						// - index.js     -> index.js
+						// - index.js     -> index.ts
```

Comment:
These are the common cases, and we also obviously handle variations like "js/App.es", "admin/js/App.es.js", and so on which exist in DXP.

Another common one seen in portal is simply "package.json", which never exists in practice at the location we're looking ("src/main/resources/META-INF/resources/"). The code in this PR will look for both "src/main/resources/META-INF/resources/package.json" and "src/main/resources/META-INF/resources/package.json.js", which is harmless.

Reply:
> Another common one seen in portal is simply "package.json" [...]

![crash](https://user-images.githubusercontent.com/905006/112185265-f8115300-8bf7-11eb-8bde-14ddc2498d47.gif)

Reply:
I'm going to take that gif as obvious tacit approval of this PR and merge it.